### DISCUSSION TITLE
Remote mouse cursor jumps when watcher changes screens

### DISCUSSION THREAD

**Discussion by danielmarschall** [rustdesk/rustdesk#6453] [Q&A] [answered]
### Bug Description

The partner computer has two screens. The person uses mouse and keyboard their usual way.

When the watcher switches between the moniors, then the mouse cursor will jump (usually to the side of the screen) which interupts their working flow.

### How to Reproduce

Switching the monitors while watching. The remote computer currently uses the mouse.

### Expected Behavior

Watching (and switching the watched monitors) should not interrupt the partner from working.

### Operating system(s) on local side and remote side

Win10 -> Win10

### RustDesk Version(s) on local side and remote side

1.2.3 -> 1.2.3

### Screenshots

(Sorry, I cannot provide a video at this moment)

**Comment by danielmarschall** [comment]
Why is this moved to the discussion section? I think when the mouse cursor jumps randomly while someone is **_watching_** (i.e. the watcher is NOT entering the remote screen area), then this is a bug in my opinion.

**Reply by danielmarschall** [reply]
@rustdesk @21pages Can you please convert this back as a bugreport? This is not a discussion, this is 100% a bug.

When I as a **_watcher_** switch the screen I am **_watching_**, then the cursor of the person working must not jump to coordinates (0, 0). This is super annoying. I would be very happy if you could fix this bug soon. Thank you very much.

**Reply by fufesou** [reply]
@danielmarschall Hi, thanks for your feedback.

You can try enable the "View mode" or disable "Keyboard" option.

<img width="623" alt="1701439367522" src="https://github.com/rustdesk/rustdesk/assets/13586388/4d368875-db46-4642-9edd-1f4e8b8f3ef9">

**Reply by danielmarschall** [reply]
@fufesou Thank you for the quick reply.
Unfortunately, I also need to control the other computer in between. Like, the other side shows and explains stuff and I watch both screens (switching between them), and sometimes I also take control and show the person how to do things.
I am aware that the remote cursor will move if my mouse cursor enters the "canvas / client area". But clicking the display icons "1" and "2" should not cause any effect on the remote computer, since my cursor is not inside the canvas / client area.

**Comment by fufesou** [comment]
Hi, would you mind using two windows like this?


https://github.com/rustdesk/rustdesk/assets/13586388/4799bb40-27c0-49a4-8b79-456fb6a53230

**Reply by fufesou** [reply]
Sometimes it is necessary to synchronize the cursor position after switching the display or pressing the menu.

Although this could be improved, not sure if this is really necessary.
Because this needs to be changed to the previous logic, position information may need to be transmitted at the same time as clicking.

**Reply by danielmarschall** [reply]
I don't have the option "Show displays in individual windows".
For me it looks like this:

![image](https://github.com/rustdesk/rustdesk/assets/28412477/e9af750d-1631-4a35-a6f6-e03f970c7095)

(and everytime I click one of these icons "1" and "2", then the cursor of the partner jumps to coordinates "0,0")

**Reply by fufesou** [reply]
Sorry, this option is introduced after 1.2.3.
It is in the nightly build.

**Reply by rustdesk** [reply]
@danielmarschall I am attempting to understand your use case. After you switches display by clicking on display icon on toolbar, where will you move your mouse? Keep mouse on toolbar?

**Reply by danielmarschall** [reply]
@rustdesk Yes, I only click the icons "1" or "2" to toggle between the two displays I am watching. My mouse does not enter the client area, it stays in the toolbar.

The use case: I have one screen, and my collegue has two screens. My collegue asks for help, so I connect to his computer. While he works and explains things to me, I need to switch the screens I am watching, because he works on two monitors. But everytime I switch the monitor, he gets interrupted, because his cursor jumps. Sometimes I will enter the client area to show something and click something, but most of the time, my mouse stays out of his screen and is only on the toolbar.

**Comment by rustdesk** [comment, marked as answer]
https://github.com/rustdesk/rustdesk/pull/6591 please try out tomorrow's  nightly build. (controlling side)

**Reply by danielmarschall** [reply]
I'm sorry for my ignorance, but I can't find how to download the nightly build. Can you show me where I find the EXE file? Thank you!

**Reply by fufesou** [reply]
https://github.com/rustdesk/rustdesk/releases/tag/nightly

[rustdesk-1.2.4-x86_64.exe](https://github.com/rustdesk/rustdesk/releases/download/nightly/rustdesk-1.2.4-x86_64.exe)

**Reply by danielmarschall** [reply]
First of all, thank you for the fast reply and the fix!


Switching the screens will no longer result in the cursor jumping at the remote user. So the user is not interrupted if I switch between screen "1" and "2" => Very good!

But now there are is a new bug:
- If the remote user moves the cursor on screen 1, then the watcher (controlling side) sees the cursor on screen 1 and on screen 2, same position. 
- If the remote user moves the cursor on screen 2, then the watcher (controlling side) does not see any cursor at all.

(Note that I only applied the update on the  controlling side. The remote side uses 1.2.3)

Please tell me if I can help with more reproduction steps, or provide more information

**Reply by rustdesk** [reply]
Sorry, I just put a fix, please try out tomorrow's nightly build (controlling side only).

**Reply by rustdesk** [reply]
Fix: https://github.com/rustdesk/rustdesk/commit/674305ce292bceee1ec1c7114997788d72f3b370

**Reply by danielmarschall** [reply]
@fufesou @rustdesk I have tested it today and it works perfect! Thank you so much for your help!
